chatgpt_wrapper/chatgpt.py

```diff
@@ -35,40 +38,67 @@ class ChatGPT:
     stream_div_id = "chatgpt-wrapper-conversation-stream-data"
     eof_div_id = "chatgpt-wrapper-conversation-stream-data-eof"
     session_div_id = "chatgpt-wrapper-session-data"
+    error_div_id = "chatgpt-wrapper-error-data"
+    lock=threading.Lock()
+    browser=None
+    page=None
+    enabled=True
+    session={}
 
     def __init__(self, headless: bool = True, browser="firefox", model="default", timeout=60, debug_log=None, proxy: Optional[ProxySettings] = None):
+        """Create an AsyncChatGPT asynchronously
+
+        Args:
+            headless (bool, optional): Start headless (no GUI) browser. Defaults to True.
+            browser (str, optional): Which browser to use. Defaults to "firefox".
+            timeout (int, optional): Timeout waiting for ChatGPT to generate message. Defaults to 60.
+            proxy (Optional[ProxySettings], optional): Network proxy. Will be passwd to playwright. Defaults to None.
+            instance (object, optional): If you already have an instance, pass it here. Used in the synchronous API. When set to None, create a new one. Defaults to None.
+
+        """
         self.log = self._set_logging(debug_log)
         self.log.info("ChatGPT initialized")
         self.play = sync_playwright().start()
         try:
             playbrowser = getattr(self.play, browser)
         except Exception:
-            print(f"Browser {browser} is invalid, falling back on firefox")
+            self.log.error(f"Browser {browser} is invalid, falling back on firefox")
             playbrowser = self.play.firefox
-        try:
-            self.browser = playbrowser.launch_persistent_context(
-                user_data_dir="/tmp/playwright",
-                headless=headless,
-                proxy=proxy,
-            )
-        except Exception:
-            self.user_data_dir = f"/tmp/{str(uuid.uuid4())}"
-            shutil.copytree("/tmp/playwright", self.user_data_dir)
-            self.browser = playbrowser.launch_persistent_context(
-                user_data_dir=self.user_data_dir,
-                headless=headless,
-                proxy=proxy,
-            )
 
-        if len(self.browser.pages) > 0:
-            self.page = self.browser.pages[0]
-        else:
-            self.page = self.browser.new_page()
-        self._start_browser()
+        if ChatGPT.browser==None:
```

Comment:
What are you using `ChatGPT` here instead of `self`? Unless I'm misunderstanding, it would limit the class to only being able to work with one browser across multiple instances of the class.

I would certainly hope it's possible in Playwright to spin up multiple browser instances, and I see no reason to limit a user of the class to only being able to use one.

Reply:
With only one browser and one tab (page), I get less risk of getting banned by OpenAI. (For example, the `Access denied` page or something). Although I'm not totally sure about this (because I made this change along with many others), but I did see more up time of my bot. 

And what's the benefit of having multiple browser instances? As long as we specify different `conversation id`s, we're having different conversations.